diff --git a/gcc/config.gcc b/gcc/config.gcc
index 53f3d3ca4..2050c6765 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1766,6 +1766,21 @@ h8300-*-linux*)
 	tmake_file="${tmake_file} h8300/t-h8300 h8300/t-linux"
 	tm_file="h8300/h8300.h elfos.h gnu-user.h linux.h glibc-stdint.h h8300/linux.h"
 	;;
+hppa*64*-*-elf*)
+	target_cpu_default="MASK_PA_11|MASK_PA_20"
+	tm_file="pa/pa64-start.h ${tm_file} elfos.h gnu-user.h \
+		 newlib-stdint.h pa/pa64-regs.h pa/pa-64.h"
+	tmake_file="${tmake_file} pa/t-pa"
+	d_target_objs="${d_target_objs} pa-d.o"
+	gas=yes gnu_ld=yes
+	;;
+hppa*-*-elf*)
+	target_cpu_default="MASK_PA_11|MASK_NO_SPACE_REGS|MASK_CALLER_COPIES"
+	tm_file="${tm_file} elfos.h gnu-user.h newlib-stdint.h \
+		 pa/pa32-regs.h"
+	tmake_file="${tmake_file} pa/t-pa"
+	d_target_objs="${d_target_objs} pa-d.o"
+	;;
 hppa*64*-*-linux*)
 	target_cpu_default="MASK_PA_11|MASK_PA_20"
 	tm_file="pa/pa64-start.h ${tm_file} elfos.h gnu-user.h linux.h \
diff --git a/libgcc/config.host b/libgcc/config.host
index 94688e623..4278b49e4 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -653,6 +653,10 @@ h8300-*-linux*)
 	tmake_file="t-linux h8300/t-linux t-softfp-sfdf t-softfp"
 	tm_file="$tm_file h8300/h8300-lib.h"
 	;;
+hppa*-*-elf*)
+	tmake_file="$tmake_file pa/t-dimode"
+	tmake_file="$tmake_file pa/t-softfp-sfdftf t-softfp"
+	;;
 hppa*64*-*-linux*)
 	tmake_file="$tmake_file pa/t-pa64-linux pa/t-dimode"
 	tmake_file="$tmake_file pa/t-softfp-sfdftf t-softfp"
