diff --git a/gas/configure.tgt b/gas/configure.tgt
index 3429f850d05..b084e571163 100644
--- a/gas/configure.tgt
+++ b/gas/configure.tgt
@@ -137,6 +137,8 @@ case ${generic_target} in
   aarch64*-*-nto*)			fmt=elf;;
   aarch64*-*-openbsd*)			fmt=elf;;
   aarch64*-*-pe* | aarch64*-*-mingw*)	fmt=coff em=pepaarch64 ;;
+
+  alpha-*-elf)				fmt=elf ;;
   alpha-*-*vms*)			fmt=evax ;;
   alpha-*-osf*)				fmt=ecoff ;;
   alpha-*-linux*ecoff*)			fmt=ecoff ;;
diff --git a/ld/configure.tgt b/ld/configure.tgt
index c62b9581d9b..bc4780a8b34 100644
--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -126,6 +126,9 @@ aarch64-*-pe* | aarch64-*-mingw*)
 			targ_extra_emuls="arm64pe"
 			targ_extra_ofiles="deffilep.o pep-dll-aarch64.o pe-dll.o pdb.o"
 			;;
+alpha*-*-elf)
+			targ_emul=elf64alpha
+			;;
 alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu)
 			targ_emul=elf64alpha_fbsd
 			targ_extra_emuls="elf64alpha alpha"

